### PR TITLE
[BridgingHeader] Provide a deterministic unique pch file path

### PR DESCRIPTION
**Pull request by cachemeifyoucan**
Teach swift driver to use a deterministic path for compiled PCH output. Also when dependency scanner is updated to return a module scanning or PCH hash from the scanning result, it will add that to the code path to have an unique PCH file path for the specific compilation.

### CODE REVIEW

**Comment by cachemeifyoucan** [comment]
@artemcm let me know what you think about this. The dependency scanner is not yet returning the context hash for main module but that can be easily added.

File: Sources/SwiftDriver/Driver/Driver.swift [hunk 1]
```diff
@@ -1779,7 +1781,7 @@ extension Driver {
   /// The swift-driver doesn't have actions, so the logic here takes the jobs and tries
   /// to mimic the actions that would be created by the C++ driver and
   /// prints them in *hopefully* the same order.
-  private func printActions(_ jobs: [Job]) {
+  private mutating func printActions(_ jobs: [Job]) {
```
Comment (artemcm) [inline Sources/SwiftDriver/Driver/Driver.swift:1784]: Why this change?

File: Sources/SwiftDriver/Driver/Driver.swift [hunk 2]
```diff
@@ -2872,23 +2874,29 @@ extension Driver {
   static func computeBridgingPrecompiledHeader(_ parsedOptions: inout ParsedOptions,
                                                compilerMode: CompilerMode,
                                                importedObjCHeader: VirtualPath.Handle?,
-                                               outputFileMap: OutputFileMap?) throws -> VirtualPath.Handle? {
+                                               outputFileMap: OutputFileMap?,
+                                               planner: ExplicitDependencyBuildPlanner?) -> VirtualPath.Handle? {
```
Comment (artemcm) [inline Sources/SwiftDriver/Driver/Driver.swift:2878]: Nit: Instead of the planner, could we pass in an optional context hash? Seems a bit overkill otherwise.

**Review by artemcm — APPROVED** [review]

**Comment by cachemeifyoucan** [comment]
@swift-ci please test

**Comment by cachemeifyoucan** [comment]
@swift-ci please test windows platform

**Comment by cachemeifyoucan** [comment]
@swift-ci please test

**Comment by shahmishal** [comment]
@swift-ci please test windows
